src/Model/Index.php

```diff
@@ -19,66 +21,118 @@
+        $this->port = $port ?? $config->getPort();
+
+        $this->_init();
+        $this->_createIndex();
+        $this->_createMapping();
+    }
 
-    )
+    /**
+     * Initialize elasticsearch client
+     */
+    private function _init()
     {
+        /** @noinspection PhpUnnecessaryFullyQualifiedNameInspection */
         $this->client = new \Elastica\Client([
-            'host' => 'elasticsearch',
-            'port' => '9200'
+            'host' => $this->host,
+            'port' => $this->port,
         ]);
+    }
+
+    /**
+     * Create index on elasticsearch server
+     *
+     * @throws \Elastica\Exception\InvalidException
+     * @throws \Elastica\Exception\ResponseException
+     */
+    private function _createIndex()
+    {
+        $this->index = $this->client->getIndex(self::INDEX_NAME);
+
+        $indexConfig = [
+            'number_of_shards'   => 4,
+            'number_of_replicas' => 1,
+            'analysis'           => [
+                'analyzer' => [
+                    'default_index'  => [
+                        'type'      => 'custom',
+                        'tokenizer' => 'standard',
+                        'filter'    => ['lowercase', 'mySnowball'],
+                    ],
+                    'default_search' => [
+                        'type'      => 'custom',
+                        'tokenizer' => 'standard',
+                        'filter'    => ['standard', 'lowercase', 'mySnowball'],
+                    ],
+                ],
+                'filter'   => [
+                    'mySnowball' => [
+                        'type'     => 'snowball',
+                        'language' => 'German',
+                    ],
+                ],
+            ],
+        ];
 
-        $this->index =  $this->client->getIndex(self::INDEX_NAME, true);
-
-        $this->index->create(
-            array(
-                'number_of_shards' => 4,
-                'number_of_replicas' => 1,
-                'analysis' => array(
-                    'analyzer' => array(
-                        'default_index' => array(
-                            'type' => 'custom',
-                            'tokenizer' => 'standard',
-                            'filter' => array('lowercase', 'mySnowball')
-                        ),
-                        'default_search' => array(
-                            'type' => 'custom',
-                            'tokenizer' => 'standard',
-                            'filter' => array('standard', 'lowercase', 'mySnowball')
-                        )
-                    ),
-                    'filter' => array(
-                        'mySnowball' => array(
-                            'type' => 'snowball',
-                            'language' => 'German'
-                        )
-                    )
-                )
-            ),
-            true
-        );
+        $this->index->create($indexConfig, true);
```

Comment:
@tomtone is this second parameter really necessary? If I understand this correctly, it means, that on every instantiation, the index will be dropped and recreated. I'd propose to remove this second parameter. Please correct me, if I'm wrong or you disagree